src/codeu/chat/DerbyStore.java

```diff
@@ -34,90 +57,155 @@ public DerbyStore() {
-						} catch (Exception e) {
-							e.printStackTrace();
-						}
 			ex.printStackTrace();
 		}
 	}
 	
-	public boolean checkForUsername(String username) throws SQLException {
-		return stmt.execute("SELECT * FROM chatuser WHERE password = '" + username + "'" );
+	public boolean checkUsernameExists(String name) throws SQLException {
+		PreparedStatement checkUserTest = conn.prepareStatement(checkUsername);
+		checkUserTest.setString(1, name);
+		ResultSet user = checkUserTest.executeQuery();
+		
+		return user == null;
 	}
 	
-	public boolean checkForPassword(String password) throws SQLException {
-		return stmt.execute("SELECT * FROM chatuser WHERE password = '" + password + "'" );
+	public String userLoginCheck(String name, String password) throws SQLException {
+		PreparedStatement checkValidUserTest = conn.prepareStatement(checkValidUser);
+		checkValidUserTest.setString(1, name);
+		checkValidUserTest.setString(2, password);
+		ResultSet user = checkValidUserTest.executeQuery();
+		
+		// TEST
+		return (user != null) ? user.getString(1) : null;
 	}
 	
 	public void addUser(User u) throws SQLException {
-		stmt.execute("INSERT INTO chatuser VALUES(" + "\'" + removeCharsInUuid(u.id.toString()) + "\'" +
-				"," + "\'" + u.name + "\'" + ", 'test', " + u.creation.inMs() + ")");
+		PreparedStatement addUser = conn.prepareStatement(addUserInfo);
+		addUser.setString(1, removeCharsInUuid(u.id.toString()));
+		addUser.setString(2, u.name);
+		addUser.setString(3, "test");
+		addUser.setLong(4, u.creation.inMs());
+		
+		addUser.executeUpdate();
+		
+		/*stmt.execute("INSERT INTO " + userTableName + " VALUES(" + "\'" + removeCharsInUuid(u.id.toString()) + "\'" +
+				"," + "\'" + u.name + "\'" + ", 'test', " + u.creation.inMs() + ")");*/
 	}
 	
 	public void addMessage(Message m) throws SQLException {
-		stmt.execute("INSERT INTO message VALUES(" + "\'" +  removeCharsInUuid(m.id.toString()) +  "\'" +
+		PreparedStatement addMessage = conn.prepareStatement(addMessageInfo);
+		addMessage.setString(1, removeCharsInUuid(m.id.toString()));
+		addMessage.setString(2, removeCharsInUuid(m.previous.toString()));
+		addMessage.setLong(3, m.creation.inMs());
+		addMessage.setString(4, removeCharsInUuid(m.author.toString()));
+		addMessage.setString(5, m.content);
+		addMessage.setString(6, removeCharsInUuid(m.next.toString()));
+		
+		addMessage.executeUpdate();
+		
+		
+		/*stmt.execute("INSERT INTO " + messageTableName + " VALUES(" + "\'" +  removeCharsInUuid(m.id.toString()) +  "\'" +
 				","  +  "\'" + removeCharsInUuid(m.previous.toString()) +  "\'" + "," +  m.creation.inMs() + "," 
-				+  "\'" + removeCharsInUuid(m.author.toString()) +  "\'" + "," +  "\'" + m.content +  "\'" + "," +  "\'" + removeCharsInUuid(m.next.toString()) +  "\'" + ")");
+				+  "\'" + removeCharsInUuid(m.author.toString()) +  "\'" + "," +  "\'" + m.content +  "\'" + "," +  "\'" + removeCharsInUuid(m.next.toString()) +  "\'" + ")");*/
```

Comment:
Same here, I'm sure you're aware of this stuff but I just want to remind you about the commented out code.